[BUG] When #define SPI_ENDSTOPS compile fails

### Description

When trying to compile marlin 2.0.x bugfix for the SKR 1.3 with LPC1768 platform and platform.io fails if i set  #define SPI_ENDSTOPS for not wiring diag to endstop. It prints 

### Steps to Reproduce

1. use platformio.ini attached 
[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/3487661/Marlin.zip)

2. Use configuration.h and configuration_adv.h attached
3. Compile

**Actual behavior:**

```
In file included from Marlin\src\inc/MarlinConfigPre.h:34,
                 from Marlin\src\inc/MarlinConfig.h:28,
                 from Marlin\src\Marlin.h:24,
                 from Marlin\src\Marlin.cpp:31:
Marlin\src\Marlin.cpp: In function 'void idle(bool)':
Marlin\src\Marlin.cpp:664:58: error: 'sg_guard_period' was not declared in this scope
     if (endstops.tmc_spi_homing.any && ELAPSED(millis(), sg_guard_period))
                                                          ^~~~~~~~~~~~~~~
Marlin\src\inc/../core/millis_t.h:28:43: note: in definition of macro 'PENDING'
 #define PENDING(NOW,SOON) ((int32_t)(NOW-(SOON))<0)
                                           ^~~~
Marlin\src\Marlin.cpp:664:40: note: in expansion of macro 'ELAPSED'
     if (endstops.tmc_spi_homing.any && ELAPSED(millis(), sg_guard_period))
                                        ^~~~~~~
*** [.pio\build\LPC1768\src\src\Marlin.cpp.o] Error 1
```

--- a/Marlin/src/Marlin.cpp
+++ b/Marlin/src/Marlin.cpp
@@ -667,8 +667,7 @@ void idle(
       #endif
     ) {
       for (uint8_t i = 4; i--;) // Read SGT 4 times per idle loop
-        if (endstops.tmc_spi_homing_check())
-          break;
+        if (endstops.tmc_spi_homing_check()) break;
     }
   #endif
 

--- a/Marlin/src/Marlin.cpp
+++ b/Marlin/src/Marlin.cpp
@@ -661,14 +661,14 @@ void idle(
 ) {
 
   #if ENABLED(SPI_ENDSTOPS)
-  if (endstops.tmc_spi_homing.any
-    #if ENABLED(IMPROVE_HOMING_RELIABILITY)
-      && ELAPSED(millis(), sg_guard_period)
-    #endif
+    if (endstops.tmc_spi_homing.any
+      #if ENABLED(IMPROVE_HOMING_RELIABILITY)
+        && ELAPSED(millis(), sg_guard_period)
+      #endif
     ) {
-    for (uint8_t i = 4; i--;) // Read SGT 4 times per idle loop
-      if (endstops.tmc_spi_homing_check())
-        break;
+      for (uint8_t i = 4; i--;) // Read SGT 4 times per idle loop
+        if (endstops.tmc_spi_homing_check())
+          break;
     }
   #endif
 

--- a/Marlin/src/Marlin.cpp
+++ b/Marlin/src/Marlin.cpp
@@ -661,9 +661,15 @@ void idle(
 ) {
 
   #if ENABLED(SPI_ENDSTOPS)
-    if (endstops.tmc_spi_homing.any && ELAPSED(millis(), sg_guard_period))
-      for (uint8_t i = 4; i--;) // Read SGT 4 times per idle loop
-        if (endstops.tmc_spi_homing_check()) break;
+  if (endstops.tmc_spi_homing.any
+    #if ENABLED(IMPROVE_HOMING_RELIABILITY)
+      && ELAPSED(millis(), sg_guard_period)
+    #endif
+    ) {
+    for (uint8_t i = 4; i--;) // Read SGT 4 times per idle loop
+      if (endstops.tmc_spi_homing_check())
+        break;
+    }
   #endif
 
   #if ENABLED(MAX7219_DEBUG)

